maliput_ros/test/maliput_ros/ros/maliput_query_node_test.cc

```diff
@@ -0,0 +1,368 @@
+                    std::chrono::nanoseconds timeout, std::chrono::nanoseconds sleep_period) {
+  return WaitForEvent(
+      node,
+      [node, service_name]() {
+        const auto service_names_and_types = node->get_service_names_and_types();
+        return service_names_and_types.end() != service_names_and_types.find(service_name);
+      },
+      timeout, sleep_period);
+}
+
+// Makes sure the transition from UNCONFIGURED via CONFIGURE is successful.
+TEST_F(MaliputQueryNodeAfterConfigurationTest, CorrectYamlConfigurationPathMakesTheConfigurationStageToPass) {
+  TransitionToConfigureFromUnconfigured();
+}
+
+// Makes sure the services are advertised when passing the configuration stage.
+TEST_F(MaliputQueryNodeAfterConfigurationTest, ConfigureStateAdvertisesServices) {
+  TransitionToConfigureFromUnconfigured();
+  ASSERT_TRUE(WaitForService(dut_, kRoadGeometryServiceName, kTimeout, kSleepPeriod));
+
+  auto service_names_and_types = dut_->get_service_names_and_types();
+
+  ASSERT_STREQ(service_names_and_types[kRoadGeometryServiceName][0].c_str(), kRoadGeometryServiceType);
+}
+
+// Makes sure services don't process the request when the node is not ACTIVE.
+TEST_F(MaliputQueryNodeAfterConfigurationTest, CallingServiceBeforeActiveYieldsToFailure) {
+  AddNodeToExecutorAndSpin(dut_);
+  TransitionToConfigureFromUnconfigured();
+  ASSERT_TRUE(WaitForService(dut_, kRoadGeometryServiceName, kTimeout, kSleepPeriod));
+
+  auto road_geometry_service = dut_->create_client<maliput_ros_interfaces::srv::RoadGeometry>(kRoadGeometryServiceName);
+  auto request = std::make_shared<maliput_ros_interfaces::srv::RoadGeometry::Request>();
+  auto future_result = road_geometry_service->async_send_request(request);
+  const auto future_status = future_result.wait_for(kTimeoutServiceCall);
+
+  ASSERT_TRUE(future_status == std::future_status::ready);
+  const auto response = future_result.get();
+  ASSERT_TRUE(response->road_geometry.id.id.empty());
+}
+
+// Makes sure the node can transtion to the ACTIVE state.
+TEST_F(MaliputQueryNodeAfterConfigurationTest, TransitionToActiveIsSuccessful) {
+  TransitionToConfigureFromUnconfigured();
+  TransitionToActiveFromConfigured();
+}
+
+// Test class used to hold the configuration of the RoadGeometryMock and validate the result of the service call.
+class TestRoadGeometryServiceCall : public MaliputQueryNodeAfterConfigurationTest {
+ public:
+  static constexpr int kSizeJunctions{0};
+  static constexpr int kSizeBranchPoints{0};
+  static constexpr double kLinearTolerance{1e-12};
+  static constexpr double kAngularTolerance{5e-12};
+  static constexpr double kScaleLength{1.};
+  const maliput::math::Vector3 kInertialToBackendFrameTranslation{1., 2., 3.};
+  const maliput::api::RoadGeometryId kRoadGeometryId{"road_geometry_id"};
+
+  void SetUp() override {
+    MaliputQueryNodeAfterConfigurationTest::SetUp();
```

Comment:
Nice chaining of SetUps and TearDowns

Reply:
BTW, missing tear down here?

Reply:
I think I don't need to do anything here because the parent classes will do it for me.